tests/client/core/reducers/test_api.js

```diff
@@ -34,6 +38,26 @@ describe('api reducer', () => {
       { ...existingState, clientApp });
   });
 
+  it('stores the userAgent', () => {
+    const existingState = { bar: 'baz' };
+    const userAgent = userAgents.firefox[1];
+    const { browser, os } = UAParser(userAgent);
+
+    assert.deepEqual(
+      api(existingState, actions.setUserAgent(userAgent)),
+      { ...existingState, userAgent, userAgentInfo: { browser, os } });
+  });
+
+  it('allows empty userAgent', () => {
+    const existingState = { bar: 'baz' };
+    const userAgent = '';
+    const { browser, os } = UAParser(userAgent);
+
+    assert.deepEqual(
+      api(existingState, actions.setUserAgent(userAgent)),
+      { ...existingState, userAgent, userAgentInfo: { browser, os } });
+  });
```

Comment:
I'd like to see a reducer test that passes in a garbage `userAgent` value. I know you have one for the component but that's not going to touch reducer code which does the actual parsing.